plugins/language_sh.lua

```diff
@@ -1,98 +1,111 @@
-    ["elif"]      = "keyword",
-    ["else"]      = "keyword",
-    ["fi"]        = "keyword",
-    ["while"]     = "keyword",
-    ["do"]        = "keyword",
-    ["done"]      = "keyword",
-    ["for"]       = "keyword",
-    ["break"]     = "keyword",
-    ["continue"]  = "keyword",
-    ["function"]  = "keyword",
-    ["local"]     = "keyword",
-    ["echo"]      = "keyword",
-    ["return"]    = "keyword",
-    ["exit"]      = "keyword",
-    ["alias"]     = "keyword",
-    ["test"]      = "keyword",
-    ["cd"]        = "keyword",
-    ["declare"]   = "keyword",
-    ["enable"]    = "keyword",
-    ["eval"]      = "keyword",
-    ["exec"]      = "keyword",
-    ["export"]    = "keyword",
-    ["getopts"]   = "keyword",
-    ["hash"]      = "keyword",
-    ["history"]   = "keyword",
-    ["help"]      = "keyword",
-    ["jobs"]      = "keyword",
-    ["kill"]      = "keyword",
-    ["let"]       = "keyword",
-    ["mapfile"]   = "keyword",
-    ["printf"]    = "keyword",
-    ["read"]      = "keyword",
+    ["case"] = "keyword",
+    ["in"] = "keyword",
+    ["esac"] = "keyword",
+    ["if"] = "keyword",
+    ["then"] = "keyword",
+    ["elif"] = "keyword",
+    ["else"] = "keyword",
+    ["fi"] = "keyword",
+    ["while"] = "keyword",
+    ["do"] = "keyword",
+    ["done"] = "keyword",
+    ["for"] = "keyword",
+    ["break"] = "keyword",
+    ["continue"] = "keyword",
+    ["function"] = "keyword",
+    ["local"] = "keyword",
+    ["echo"] = "keyword",
+    ["return"] = "keyword",
+    ["exit"] = "keyword",
+    ["alias"] = "keyword",
+    ["test"] = "keyword",
+    ["cd"] = "keyword",
+    ["declare"] = "keyword",
+    ["enable"] = "keyword",
+    ["eval"] = "keyword",
+    ["exec"] = "keyword",
+    ["export"] = "keyword",
+    ["getopts"] = "keyword",
+    ["hash"] = "keyword",
+    ["history"] = "keyword",
+    ["help"] = "keyword",
+    ["jobs"] = "keyword",
+    ["kill"] = "keyword",
+    ["let"] = "keyword",
+    ["mapfile"] = "keyword",
+    ["printf"] = "keyword",
+    ["read"] = "keyword",
     ["readarray"] = "keyword",
-    ["pwd"]       = "keyword",
-    ["select"]    = "keyword",
-    ["set"]       = "keyword",
-    ["shift"]     = "keyword",
-    ["source"]    = "keyword",
-    ["time"]      = "keyword",
-    ["type"]      = "keyword",
-    ["until"]     = "keyword",
-    ["unalias"]   = "keyword",
-    ["unset"]     = "keyword",
-    ["true"]      = "literal",
-    ["false"]     = "literal"
-  }
+    ["pwd"] = "keyword",
+    ["select"] = "keyword",
+    ["set"] = "keyword",
+    ["shift"] = "keyword",
+    ["source"] = "keyword",
+    ["time"] = "keyword",
+    ["type"] = "keyword",
+    ["until"] = "keyword",
+    ["unalias"] = "keyword",
+    ["unset"] = "keyword",
+    ["true"] = "literal",
+    ["false"] = "literal",
+  },
 }
 
+local function merge_tables(a, b) for _, v in pairs(b) do table.insert(a, v) end end
+
+merge_tables(string_interpolation_syntax.patterns, {
+  { pattern = "%$[%w_]+", type = "keyword2" },
```

Comment:
Other escape sequences that should be taken into consideration:
https://www.baeldung.com/linux/bash-escape-characters#3-ansi-c-combinations

Reply:
I have added a special syntax for `$''` strings and `\u`/`\x` sequences.

I also added the `\\.` pattern but I think it would be nice to be more specific and highlight only characters that actually get escaped but I have a hard time figuring out which ones to highlight since things like `\\` are escaped by bash but things like `\n` aren't always.

But if you use something like `\p` you probably meant to escape it so it should be highlighted.

Reply:
I tested escape sequences on single quote strings without the $ sign and they also work eg: `'\n'`, what doesn't works on single quotes is the expressions expansion (interpolation).

Reply:
oh yes, it is a zsh thing :) ignore what I said. Anyways, tested, looks good so merged into https://github.com/pragtical/plugins